### PR TITLE
Modernize & TypeScript-ify "Range Manipulation" sample

### PR DESCRIPTION
**Pull request by Zlatkovsky**

### CODE REVIEW

**Comment by Zlatkovsky** [comment]
@jakobpn or @kbrandl , could you please take a look and approve?

File: samples/word/04-range/split-words-of-first-paragraph.yaml [hunk 1]
```diff
@@ -4,157 +4,84 @@ description: 'Show how to split a paragraph into word ranges and then traverses
...
-            }
+        function pause(milliseconds) {
+            return new Promise(resolve => setTimeout(resolve, milliseconds));
         }
 
-        function setup() {
-            Word.run(function (context) {
+        async function setup() {
+            await Word.run(async (context) => {
                 // lets insert a couple of paragraphs to illustrate the point..
```
Comment (kbrandl) [inline samples/word/04-range/split-words-of-first-paragraph.yaml]: I realize you didn't write this comment, but can you edit it? :)   

- remove **lets**
- capitalize **Insert**
- remove the duplicate period from the end of the sentence

File: samples/word/04-range/split-words-of-first-paragraph.yaml [hunk 2]
```diff
@@ -4,157 +4,84 @@ description: 'Show how to split a paragraph into word ranges and then traverses
...
+        $("#setup").click(() => tryCatch(setup));
+        $("#highlight").click(() => tryCatch(highlightWords));
+
+        async function highlightWords() {
+            await Word.run(async (context) => {
+                let paragraph = context.document.body.paragraphs.getFirst();
+                let words = paragraph.split(
+                    [" "], true /* trimDelimiters*/, true /* trimSpaces */);
+                words.load("text");
+                await context.sync();
```
Comment (kbrandl) [inline samples/word/04-range/split-words-of-first-paragraph.yaml]: In all the Excel snippets I've seen, **await context.sync();** always has a blank line before it and after it (presumably to call attention to the fact that we're making an async call).  Perhaps follow that convention here as well?

File: samples/word/04-range/split-words-of-first-paragraph.yaml [hunk 3]
```diff
@@ -4,157 +4,84 @@ description: 'Show how to split a paragraph into word ranges and then traverses
...
+
+                for (let i = 0; i < words.items.length; i++) {
+                    if (i >= 1) {
+                        words.items[i - 1].font.highlightColor = "#FFFFFF";
+                    }
+                    words.items[i].font.highlightColor = "#FFFF00";
+                    await context.sync();
```
Comment (kbrandl) [inline samples/word/04-range/split-words-of-first-paragraph.yaml]: Same comment as before -- perhaps follow convention of including blank line before and after **await context.sync();**?

File: samples/word/04-range/split-words-of-first-paragraph.yaml [hunk 4]
```diff
@@ -4,157 +4,84 @@ description: 'Show how to split a paragraph into word ranges and then traverses
...
-        function setup() {
-            Word.run(function (context) {
+        async function setup() {
+            await Word.run(async (context) => {
                 // lets insert a couple of paragraphs to illustrate the point..
                 context.document.body.clear();
                 context.document.body.insertParagraph("Video provides a powerful way to help you prove your point. When you click Online Video, you can paste in the embed code for the video you want to add. You can also type a keyword to search online for the video that best fits your document.", "start");
                 context.document.body.paragraphs.getLast().insertText("To make your document look professionally produced, Word provides header, footer, cover page, and text box designs that complement each other. For example, you can add a matching Online cover page, header, and sidebar. Click Insert and then choose the Online elements you want from the different Online galleries.", "replace");
-                return context.sync();
+                await context.sync();
```
Comment (kbrandl) [inline samples/word/04-range/split-words-of-first-paragraph.yaml]: Same comment as before -- perhaps follow convention of including blank line before and after **await context.sync();**?

**Pull request merged**